nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/flow/synchronization/StandardVersionedComponentSynchronizer.java

```diff
@@ -2739,6 +2742,41 @@ private void updateRemoteProcessGroup(final RemoteProcessGroup rpg, final Versio
         }
     }
 
+    private void synchronizeRemoteGroupPorts(final Set<RemoteGroupPort> remoteGroupPorts, final Set<VersionedRemoteGroupPort> proposedPorts) {
+        final Map<String, VersionedRemoteGroupPort> inputPortsByTargetId = mapRemoteGroupPortsByTargetId(proposedPorts);
+        remoteGroupPorts.forEach(port -> {
+            final VersionedRemoteGroupPort proposedPort = inputPortsByTargetId.get(port.getTargetIdentifier());
+            if (proposedPort != null) {
+                if (proposedPort.getBatchSize() != null) {
+                    final BatchSize batchSize = proposedPort.getBatchSize();
+                    if (batchSize.getSize() != null) {
+                        port.setBatchSize(batchSize.getSize());
+                    }
+                    if (batchSize.getCount() != null) {
+                        port.setBatchCount(batchSize.getCount());
+                    }
+                    if (batchSize.getDuration() != null) {
```

Comment:
I did some testing. Created a flow that used an RPG with default batch config. Saved as v1. Then updated the batch size and duration but left count as `null`. Saved as v2. Then changed to v1. It left the Size as "10 KB" and the duration of "1 sec". And now it shows me in a Conflicted state because the flow differs from v1.
We can call `port.setBatchSize`, `port.setBatchCount`, and `port.setDuration` with `null` values so I think we need to eliminate these null checks here. We still need them for the `useCompression` and `maxConcurrentTasks`, just not for the batch config.

Reply:
Good call -- that totally makes sense.  Thanks!

Reply:
Ok, I made that update and tested as you described, and it appears to work correctly now.